craft_parts/overlays/layers.py

```diff
@@ -98,3 +101,51 @@ def save(self, part: Part) -> None:
     def hex(self) -> str:
         """Return the current hash value as a hexadecimal string."""
         return self.digest.hex()
+
+
+class LayerStateManager:
+    """An in-memory layer state management helper for action planning.
+
+    :param part_list: The list of parts in the project.
+    """
+
+    def __init__(self, part_list: List[Part], base_layer_hash: Optional[LayerHash]):
+        self._part_list = part_list
+        self._base_layer_hash = base_layer_hash
+
+        self._layer_hash: Dict[str, Optional[LayerHash]] = dict()
```

Comment:
Create the empty dict with `{}`.

Reply:
It seems that `dict()` could lead to gratuitous inefficiency, so replacing it with `{}`.

Reply:
I really dislike this reality as `dict()` is much easier to quickly gloss over than `{}`.